### ISSUE TITLE
EC2MetadataUtils.getItems() takes a long time to fail outside of EC2

### ISSUE THREAD

**Issue by tfeak** [aws/aws-sdk-java#799]
This issue cropped up between version 1.11.15 and 1.11.16

When running my server locally on my desktop I now get ~30 second timeouts before proceeding. Previously, this didn't occur when the host for the EC2Metadata Service was unavailable.

This is cropping up in a couple places in our server startup, which results in a 1.5 minute server startup instead of a 10-15 second start time. I grant this probably isn't an issue inside of EC2, but it definitely impacts development time.

Is there any way for this for this lookup to fail more rapidly when called outside of EC2?

**Comment by tfeak** [comment]
A bit more digging looks like the code change between 1.11.15 and 1.11.16 was actually a switch to using EC2CredentialsUtils.readResource() method instead of EC2MetadataClient.readResource() which appear to be quite different.

**Comment by varunnvs92** [comment]
@tfeak 
I have identified the issue. In 1.11.16, we started retrying when a connection to the end host cannot be opened. We retry 5 times before we fail, so that explains the increase in startup time. 
https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/internal/EC2CredentialsUtils.java#L61-L96

I will fix this and include it in our next release. Will update the thread once the fix has been pushed.
